pkg/cri/server/sandbox_run.go

```diff
@@ -139,33 +162,18 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 			return nil, fmt.Errorf("failed to create network namespace for sandbox %q: %w", id, err)
 		}
 		sandbox.NetNSPath = sandbox.NetNS.GetPath()
-		defer func() {
-			if retErr != nil {
-				deferCtx, deferCancel := ctrdutil.DeferContext()
-				defer deferCancel()
-				// Teardown network if an error is returned.
-				if err := c.teardownPodNetwork(deferCtx, sandbox); err != nil {
-					log.G(ctx).WithError(err).Errorf("Failed to destroy network for sandbox %q", id)
-				}
 
+		defer func() {
```

Comment:
I thought we discussed to move this down after container creation, and update the container net NS afterwards. https://github.com/containerd/containerd/pull/5904#pullrequestreview-966050936

Are we going to do that in this change?

Reply:
Sorry for the confusion. I tried to do that way but found a problem that `sandboxContainerSpec` function needs the netns path when creating the container. I understand we can add a new function, like `updateContainerSpecNetNS` for linux and windows but I want to save the time for now.

Also creating NetNS first is still correct.

Reply:
ok, I added a new function to update the nspath after container is created.